Hardcore.lua

```diff
@@ -552,6 +515,43 @@ function Hardcore:PLAYER_DEAD()
 	end
 end
 
+function Hardcore:PLAYER_TARGET_CHANGED()
+	if Hardcore_Character.grief_warning_conditions == GRIEF_WARNING_BOTH_FACTIONS then
+		if UnitIsPVP("target") and UnitGUID("target") ~= PLAYER_GUID then
+			local faction, _ = UnitFactionGroup("target")
+			if faction ~= nil then
+				if faction ~= PLAYER_FACTION then
+					local target_name, _ = UnitName("target")
+					Hardcore:ShowAlertFrame(ALERT_STYLES.hc_pvp_warning, "Target " .. target_name .. " is PvP enabled!")
+				elseif UnitPlayerControlled("target") then
+					local target_name, _ = UnitName("target")
+					Hardcore:ShowAlertFrame(ALERT_STYLES.hc_pvp_warning, "Target " .. target_name .. " is PvP enabled!")
+				end
+			end
+		end
```

Comment:
This can be reduced by 6 lines pretty cleanly:
```
if faction and UnitPlayerControlled("target")  then
	local target_name, _ = UnitName("target")
	Hardcore:ShowAlertFrame(ALERT_STYLES.hc_pvp_warning, "Target " .. target_name .. " is PvP
end
```

Reply:
you mean this entire chunk?

Reply:
Lines 522-531

Reply:
Hmm, I had intended to keep enemy faction npc's on the alert, so that logic would have to branch

Reply:
I updated it for an `or` condition, does that work?

Reply:
The main issue was that there were multiple logic branches for the same outcome

Reply:
Not sure i follow; original logic should be 

If same faction and player controlled then warn

If enemy faction then warn

Reply:
Try this:

If (same faction and player controlled) or enemy faction then warn

Reply:
Oh gotcha, I was looking at the suggestion above